apluslms_roman/utils/path_mapping.py

```diff
@@ -0,0 +1,48 @@
+import json
+import logging
+import re
+from os import environ
+from pathlib import PurePosixPath
+
+logger = logging.getLogger(__name__)
+json_re = re.compile(r'^(?:["[{]|(?:-?[1-9]\d*(?:\.\d+)?|null|true|false)$)')
+
+
+def get_host_path(original, mapping):
+    ret = original
+    orig_path = PurePosixPath(original)
+    for k, v in mapping.items():
+        try:
+            logger.debug("Mapping:%s:%s", k, v)
+            relative_path = orig_path.relative_to(k)
+            ret = PurePosixPath(v).joinpath(relative_path)
+            return str(ret)
+        except ValueError:
+            logger.warning("Error when composing new path!")
+            pass
+    return str(ret)
+
+
+def nest_dict(flat_dict, sep):
```

Comment:
This could be implemented as (and moved into load_from_env):

```
from apluslms_yamlidator.document import find_ml

nested = {}
for keys, value in flat_dict.items():
    dict_, key = find_ml(nested, keys, create_dicts=True)
    dict_[key] = value
return nested
```

Sadly, that doesn't support `sep`, but I don't see there would bee ever need to support other than `.`.

ref: https://github.com/apluslms/roman/blob/milestone0.3/apluslms-yamlidator/apluslms_yamlidator/document.py#L30

Reply:
I might end up extending the find_ml to support key strings like `foo[].bar[]` some day :)